Adds PowerMetric benchmark to measure Light vs Dark theme

yenerm opened the pull request:
Adds PowerMetric benchmark which scrolls through the topics list in Light vs Dark theme

dturner commented on benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/ScrollTopicListPowerMetricsBenchmark.kt:
```diff
@@ -0,0 +1,82 @@
+                pressHome()
+                startActivityAndWait()
+                allowNotifications()
+                // Navigate to interests screen
```
The following code doesn't appear to navigate to the interests screen, it just opens the app and clicks the Settings button. Either the comment is wrong or the code needs to be updated.

dturner commented on benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/InterestsActions.kt:
```diff
@@ -48,3 +48,17 @@ fun MacrobenchmarkScope.interestsToggleBookmarked() {
     device.waitForIdle()
 }
+
+fun MacrobenchmarkScope.setAppTheme(isDark: Boolean) {
```
I'm not sure why this function is inside `InterestsActions`. The Settings button can be found on every screen in the app inside the top app bar. It might make more sense to have this in a `ForYouActions` file (because ForYou is the first screen in the app), or just `AppActions`.

dturner commented on benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/InterestsActions.kt:
```diff
@@ -48,3 +48,17 @@ fun MacrobenchmarkScope.interestsToggleBookmarked() {
+    device.waitForIdle()
+    device.findObject(By.text("OK")).click()
+
+    // Wait until interests are shown on screen
```
It looks like this code is waiting for the top app bar to be shown, rather than any interests/topics.

dturner commented on benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/ScrollTopicListPowerMetricsBenchmark.kt:
```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 The Android Open Source Project
```
nit: new files should have the current year for copyright

dturner reviewed (changes requested):
I'm not sure if this is relevant but the way the app looks is defined by: 

- the brand which the user chooses, this can be default or Android green
- if the default brand is chosen then this will be pink on API < 31 or will use dynamic color on API 31+ if the user has selected that option  
- lastly, once the brand has been chosen, dark mode determines the actual colours used. 

I'm guessing you just want to test light vs dark mode energy consumption but there might be some variance with the brand configurations as well.

yenerm commented:
Thanks Don! I addressed the comments. Also good points on variance with the brand configurations. I think the benchmark still makes sense as a sample in NiA. 
Eventually (no the first priority) I am planning to add similar sample to performance samples but it doesn't currently have a theme for dark mode. Let me know wdyt?

dturner commented:
Totally fine for this to be in this project. Implementing the theme wasn't trivial so probably makes sense to benchmark it here, rather than introducing that complexity into a simpler sample just to test it.
